cli/src/main/java/org/infinispan/cli/connection/rest/RestConnection.java

```diff
@@ -216,499 +167,72 @@ private RestResponse handleResponseStatus(RestResponse response) throws IOExcept
-                     break;
-                  }
-               }
-               break;
-            }
-            case Stats.CMD: {
-               Resource resource = activeResource;
-               if (command.hasArg(CliCommand.NAME)) {
-                  resource = pathToResource(command.arg(CliCommand.NAME));
-               }
-               if (resource instanceof CacheResource) {
-                  response = client.cache(resource.getName()).stats();
-               } else if (resource instanceof ContainerResource) {
-                  response = client.cacheManager(resource.getName()).stats();
-               } else {
-                  String name = resource.getName();
-                  throw MSG.invalidResource(name.isEmpty() ? "/" : name);
-               }
-               break;
-            }
-            case Logging.CMD: {
-               switch (command.arg(Logging.TYPE)) {
-                  case Logging.Loggers.CMD: {
-                     response = client.server().logging().listLoggers();
-                     break;
-                  }
-                  case Logging.Appenders.CMD: {
-                     response = client.server().logging().listAppenders();
-                     break;
-                  }
-                  case Logging.Set.CMD: {
-                     if (command.hasArg(Logging.Set.APPENDERS)) {
-                        List<String> appenders = command.argAs(Logging.Set.APPENDERS);
-                        response = client.server().logging().setLogger(command.arg(Logging.NAME), command.option(Logging.Set.LEVEL), appenders.toArray(new String[0]));
-                     } else {
-                        response = client.server().logging().setLogger(command.arg(Logging.NAME), command.option(Logging.Set.LEVEL));
-                     }
-                     break;
-                  }
-                  case Logging.Remove.CMD: {
-                     response = client.server().logging().removeLogger(command.arg(Logging.NAME));
-                     break;
-                  }
-               }
-               break;
-            }
-            case Server.CMD: {
-               switch (command.arg(Logging.TYPE)) {
-                  case Server.Report.CMD: {
-                     responseMode = ResponseMode.FILE;
-                     response = client.server().report();
-                  }
-                  break;
+            break;
+         case FILE:
+            String contentDisposition = parseHeaders(r).get("Content-Disposition").get(0);
+            String filename = contentDisposition.split("filename=")[1];
+            Path file = workingDir.resolve(filename);
+
+            try (OutputStream os = Files.newOutputStream(file); InputStream is = parseBody(r, InputStream.class)) {
+               byte[] buffer = new byte[8 * 1024];
+               int bytesRead;
+               while ((bytesRead = is.read(buffer)) != -1) {
```

Comment:
with response code `204`, `is` is `null` and it will throw an NPE. You probably want better handling :)

Reply:
We don't have resources which return a FILE which also set the status to 204.